tmpdir.go

```diff
@@ -26,3 +28,69 @@ func Setup(path string) (err error) {
 func Cleanup() error {
 	return os.RemoveAll(Dir)
 }
+
+// MkTemp creates a unique filename in the temporary directory and returns a
+// writable file descriptor. This function is safe for any arbitrary filename.
+//
+// Subdirectories are not supported. The Setup() function must be called first.
+func MkTemp(filename string) (*os.File, error) {
+	if Dir == "" {
+		panic("Dir is empty. Setup() needs to be called first.")
```

Comment:
Should we not just return an error here?

Reply:
Ehm, possibly? It's actually from when it just returned a `string` and didn't return an error.

But it's a clear programming/state error, and seems to be an appropriate use of panic to me.

Reply:
Yeah, that's the same pattern we use with `Must`.  I think it makes sense.

Reply:
Not a recoverable error

Reply:
It may not be recoverable, but it can be handled more gracefully than a panic.

My 2¢:  Use `MkTemp` for things that are done during requests, etc, and `MustMkTemp` for things that happen during start-up, etc.

But it's a bikeshedding discussion, so I don't really care :P

Reply:
> MustMkTemp for things that happen during start-up, etc.

I don't know if we ever use this; I think not. I just changed it to an error now. We can always add `Must*()` later of we actually want to use it.